libs/fujitsu-thermal-printer/src/mocks/file_printer.ts

```diff
@@ -0,0 +1,183 @@
+  __dirname,
+  '../../dev-workspace'
+);
+
+function getMockPrinterPath(): string {
+  // istanbul ignore next
+  if (process.env.NODE_ENV === 'development') {
+    return DEV_MOCK_FUJITSU_PRINTER_DIR;
+  }
+
+  return DEFAULT_MOCK_FUJITSU_PRINTER_DIR;
+}
+
+type MockStateFileContents = PrinterStatus;
+
+function getMockPrinterStateFilePath(): string {
+  return join(getMockPrinterPath(), MOCK_FUJITSU_PRINTER_STATE_FILENAME);
+}
+
+function getMockPrinterOutputPath(): string {
+  return join(getMockPrinterPath(), MOCK_FUJITSU_PRINTER_OUTPUT_DIRNAME);
+}
+
+/**
+ * Converts a MockFileContents object into a Buffer
+ */
+function serializeMockFileContents(
+  mockStateFileContents: MockStateFileContents
+): Buffer {
+  return Buffer.from(JSON.stringify(mockStateFileContents), 'utf-8');
+}
+
+/**
+ * Converts a Buffer created by serializeMockFileContents back into a MockFileContents object
+ */
+function deserializeMockFileContents(file: Buffer): MockStateFileContents {
+  return JSON.parse(file.toString('utf-8'));
+}
+
+function writeToMockFile(mockStateFileContents: MockStateFileContents): void {
+  mkdirSync(getMockPrinterPath(), { recursive: true });
+  writeFileSync(
+    getMockPrinterStateFilePath(),
+    serializeMockFileContents(mockStateFileContents)
+  );
+  // Create the output directory whenever we write to the state file, so that
+  // it's there before any prints are created
+  mkdirSync(getMockPrinterOutputPath(), { recursive: true });
+}
+
+export function initializeMockFile(): void {
+  writeToMockFile({
+    state: 'idle',
+  });
+}
+
+/**
+ * A helper for readFromMockFile. Returns undefined if the mock file doesn't exist or can't be
+ * parsed.
+ */
+function readFromMockFileHelper(): Optional<MockStateFileContents> {
+  const mockFilePath = getMockPrinterStateFilePath();
+  if (!existsSync(mockFilePath)) {
+    return undefined;
+  }
+  const file = readFileSync(mockFilePath);
+  try {
+    return deserializeMockFileContents(file);
+  } catch {
+    /* istanbul ignore next */
+    return undefined;
+  }
+}
+
+/**
+ * Reads and parses the contents of the file underlying a MockFileUsbDrive
+ */
+function readFromMockFile(): MockStateFileContents {
+  let mockFileContents = readFromMockFileHelper();
+  if (!mockFileContents) {
+    initializeMockFile();
+    mockFileContents = readFromMockFileHelper();
+    assert(mockFileContents !== undefined);
+  }
+  return mockFileContents;
+}
+
+export const FILE_FUJITSU_PRINTER_POLLING_INTERVAL_MS = 1000;
+export const FILE_FUJITSU_PRINTER_POLLING_COUNT = 5;
+
+export class MockFileFujitsuPrinter implements FujitsuThermalPrinterInterface {
+  private readonly printPollingInterval: number;
+
+  constructor(printPollingInterval = FILE_FUJITSU_PRINTER_POLLING_INTERVAL_MS) {
+    this.printPollingInterval = printPollingInterval;
+  }
+
+  getStatus(): Promise<PrinterStatus> {
+    return Promise.resolve(readFromMockFile());
+  }
+
+  async print(data: Uint8Array): Promise<PrintResult> {
```

Comment:
This is more elaborate than mocks usually should be - it actually sleeps and introduces real delay to similar printing. This could present issues testing in an integration setting in the future. HOWEVER I think that it's really valuable to have the pause because it allows the developer to change printer state mid-print, which is the only way to enter some software states.